src/pynxtools/nomad/parser.py

```diff
@@ -491,18 +491,25 @@ def parse(
                 app_sec = getattr(self.nx_root, app)
                 try:
                     app_entry = getattr(app_sec, "ENTRY")
+                    if len(app_entry) < 1:
+                        raise AttributeError()
                 except AttributeError:
                     app_entry = getattr(app_sec, "entry")
+                    if len(app_entry) < 1:
+                        raise AttributeError()
```

Comment:
Could we make this a bit cleaner? Also would be nice to have some information in the error message.
```suggestion
                for attr_name in ("ENTRY", "entry"):
                    app_entry = getattr(app_sec, attr_name, None)
                    if app_entry and len(app_entry) >= 1:
                        break
                    else:
                        raise AttributeError("No valid entry found in the NeXus file.")
```

Reply:
The suggested code would raise exception even if ENTRY is not found, but we need to quit only if neither 'ENTRY' nor 'entry' is present and even in this case we want to catch this exception as it is not a real error at this point, but only an indication for what type of Experiment label to put into 'entry_type'.

Reply:
Ok, let's keep your version then.